pogs/interface_test.go

```diff
@@ -0,0 +1,168 @@
+package pogs
+
+import (
+	"golang.org/x/net/context"
+	"testing"
+	"zombiezen.com/go/capnproto2"
+	air "zombiezen.com/go/capnproto2/internal/aircraftlib"
+)
+
+type simpleEcho struct{}
+
+func checkFatal(t *testing.T, name string, err error) {
+	if err != nil {
+		t.Fatalf("%s for TestInsertIFace: %v", name, err)
+	}
+}
+
+func (s simpleEcho) Echo(p air.Echo_echo) error {
+	text, err := p.Params.In()
+	if err != nil {
+		return err
+	}
+	p.Results.SetOut(text)
+	return nil
+}
+
+type EchoBase struct {
+	Echo air.Echo
+}
+
+type EchoBases struct {
+	Bases []EchoBase
+}
+
+type Hoth struct {
+	Base EchoBase
+}
+
+func TestInsertIFace(t *testing.T) {
```

Comment:
@zombiezen, I think I've addressed everything besides this point. Any thoughts on my comment above?